paymentsheet/src/main/java/com/stripe/android/paymentelement/embedded/EmbeddedConfirmationHelper.kt

```diff
@@ -0,0 +1,72 @@
+package com.stripe.android.paymentelement.embedded
+
+import androidx.activity.result.ActivityResultCaller
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import com.stripe.android.common.model.asCommonConfiguration
+import com.stripe.android.lpmfoundations.paymentmethod.PaymentMethodMetadata
+import com.stripe.android.paymentelement.EmbeddedPaymentElement
+import com.stripe.android.paymentelement.ExperimentalEmbeddedPaymentElementApi
+import com.stripe.android.paymentelement.confirmation.ConfirmationHandler
+import com.stripe.android.paymentelement.confirmation.toConfirmationOption
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.state.PaymentElementLoader
+import kotlinx.coroutines.launch
+
+@ExperimentalEmbeddedPaymentElementApi
+internal class EmbeddedConfirmationHelper(
```

Comment:
I want to keep the view model as small as possible, and primarily act as an integration point. I pulled this out of it to allow for easy unit testing of this functionality, and create a clear boundary of what state is used for confirmation.